pre_commit_hooks/forbid_new_submodules.py

```diff
@@ -1,14 +1,26 @@
-from typing import Optional
+import os
+import sys
 from typing import Sequence
 
 from pre_commit_hooks.util import cmd_output
 
 
-def main(argv: Optional[Sequence[str]] = None) -> int:
-    # `argv` is ignored, pre-commit will send us a list of files that we
-    # don't care about
+def main(argv: Sequence[str] = sys.argv[1:]) -> int:
```

Comment:
this sets a mutable default value which does not work as you would expect in python -- please don't change this, the original argument handling was correct

```pycon
>>> def f(x=[]):
...     x.append('wat')
...     print(x)
... 
>>> f(['hi'])
['hi', 'wat']
>>> f(['hi'])
['hi', 'wat']
>>> f()
['wat']
>>> f()
['wat', 'wat']
>>> f()
['wat', 'wat', 'wat']
>>> f()
['wat', 'wat', 'wat', 'wat']
>>> f()
['wat', 'wat', 'wat', 'wat', 'wat']
```

Reply:
Yep, I'm aware of the behavior of mutable defaults in python. In this case it's not a problem because the function will always be called only once and without arguments by setuptools machinery. However, I agree that using it may be fragile, for example, in tests.